addons/account/static/src/js/section_and_note_fields_backend.js

```diff
@@ -94,8 +118,11 @@ var SectionAndNoteFieldOne2Many = FieldOne2Many.extend({
 // We want a FieldChar for section,
 // and a FieldText for the rest (product and note).
 var SectionAndNoteFieldText = function (parent, name, record, options) {
-    var isSection = record.data.display_type === 'line_section';
-    var Constructor = isSection ? FieldChar : ListFieldText;
+    const isPurchaseOrderLine = record.model === 'purchase.order.line';
```

Comment:
Why this distinction with purchase.order.line ?
If required, it’d be better to control this behaviour with a widget option rather than hardcoding the model name

Reply:
Because it has been considered that the feature wasn't needed/useful for the purchase scope ^^.
Indeed, the current implementation is kind of a hack right now, the code should be cleaned up before merge (and some bugs remain, waiting for a fix :D)